server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java

```diff
@@ -273,7 +273,7 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
    protected ExecutorService getExecutor(String threadPrefix) {
       if (this.executor == null || this.executor.isShutdown()) {
          DefaultThreadFactory factory = new DefaultThreadFactory(threadPrefix + "-ServerHandler");
-         int workerThreads = Integer.getInteger(WORKER_THREADS_SYS_PROP, configuration.workerThreads());
+         int workerThreads = getWorkerThreads();
```

Comment:
Hey, this should not use the same number as netty threads. This should still be able to grow to hundreds. In case that you have a persistence layer, you'll need those.

Reply:
OK, so what about configuring the Netty event loop threads as 'ioThreads' and using `workerThreads` for the blocking ops?

Looking into REST it seems that the operation is invoked directly on the netty thread, which is a problem. (how is it possible that we haven't seen scalability issues in perfack?) @slaskawi can you confirm?

Reply:
ioThreads is much better than slaveThreads. I'll use that. PerfAck doesn't use persistence, so maybe the impact is not as bad ?

Reply:
Hot Rod moves the execution to different thread pool with persistence because the `putAsync` it executes actually can be blocking (if this node is a primary owner). RPC is non-blocking internally so it does not become an issue.
However REST seems to run everything using synchronous `put` (unless special headers are included) and that means that the Netty thread will be blocked until we do the roundtrip to other owners.